src/model.cpp

```diff
@@ -135,7 +135,10 @@ Model::Model(const int nx_rdata, const int nxtrue_rdata, const int nx_solver,
                              ndxdotdp_explicit, CSC_MAT);
         dxdotdp_implicit =
             SUNMatrixWrapper(nx_solver, static_cast<int>(p.size()),
-                             ndwdp + ndxdotdw, CSC_MAT);
+                             ndwdp + ndxdotdw, CSC_MAT); // guess size
```

Comment:
Does this need to be an upper bound, as it is used for capacity?

Reply:
No, reallocate will allocate more space if necessary.